duo.c

```diff
@@ -179,7 +179,7 @@ _sign_request(struct duo_ctx *ctx, const char *method, const char *uri,
            return (NULL);
         }
 
-        HMAC_Init_ex(hmac, ctx->skey, strlen(ctx->skey), EVP_sha1(), NULL);
+        HMAC_Init_ex(hmac, ctx->skey, strlen(ctx->skey), EVP_sha512(), NULL);
```

Comment:
does line 169 need to change also?  IIRC SHA2 digests are longer, so I don't know if MD will be big enough (because I don't know what SHA_DIGEST_LENGTH is, it's probably coming from one of the includes)

Reply:
yeah I think you need SHA512_DIGEST_LENGTH, see https://github.com/openssl/openssl/blob/b6298a7f8de45b287bfbad5de282fbceb68abb02/include/openssl/sha.h#L87